luaa.c

```diff
@@ -411,6 +412,126 @@ get_modifier_name(int map_index)
     return 0; /* \0 */
 }
 
+// Helper function for luaA_get_key_name() below.
+// Will return the UTF-32 codepoint IF AND ONLY IF the input is exactly one
+// valid UTF-8 character. Otherwise, it will return zero.
+static uint32_t
+one_utf8_to_utf32(const char* input, const size_t length) {
+    // When casting a char to a uint32_t, the missing bits will all be set to
+    // the highest bit in the original char. Thus the exaggerated masking of
+    // every value (Unicode is defined around bytes)
+    if (length < 1 || length > 4)
+        return 0;
+    if (length == 1)
+    {
+        if ((*input & 0x80) == 0)
+            return *input;
+        else
+            return 0;
+    }
+    uint32_t mask = (0xff << (7 - length)) & 0x000000ff;
+    const uint32_t check = (mask << 1) & 0x000000ff;
+    uint32_t ucs = 0x00000000;
+    if ((*input & mask) == check)
+    {
+        mask = (~mask) & 0x000000ff;
+        ucs = (*input & mask) << (6 * (length-1));
+    }
+    else
+        return 0;
+    for (size_t i = 1; i < length; i++)
+    {
+        if ((*(input + i) & 0x000000c0) != 0x00000080)
+            return 0;
+        ucs += (*(input + i) & 0x0000003f) << (6 * (length-i-1));
+    }
+    return ucs;
+}
+
+/** Get Xlib keysym and xkb_keysym_to_uf8 from an Awesome keycode.
+ *
+ * xkb_keysym_to_utf8 returns a UTF-8 representation of the symbol a given
+ * symkey generates. It matches the engraving of the key for level-0 symbols
+ * (but lowercase)
+ *
+ * Keycodes may be given in a string in any valid format for `awful.key`:
+ * "#" + keycode, the symkey name and the UTF-8 representation will all work.
+ *
+ * If no suitable keysym is found, or a malformed keycode is given as an
+ * argument, this function will return (nil, nil)
```

Comment:
Actually... it just does not return anything, I think. That's a difference. Is it worth documenting?

Reply:
It's not a difference for Lua.

`return 0;` passes 0 values from the stack to Lua, and Lua reports `nil` if you ask too many values from a function.

Meaning that you're right about your suggestion for lines 518 and 519: `lua_pushnil(L); lua_pushnil(L); return 2;` has the same effect as `return 0;`.

Reply:
Hm... I was sure there was a way to tell apart `nil` and 'nothing', but I cannot find a way to do so from Lua, only from C.

Reply:
Oh and: Sorry for making a useless suggestion then.

Reply:
Hey, don't be. _Every other one_ has been a good suggestion.